### PR TITLE
Config load

### LINKED ISSUE #387: Neuron config does not work.
https://github.com/opentensor/bittensor/blob/master/miners/text/gpt2_exodus.py#L111

--neuron.config should load defaults into the miner which overwrites the defaults passed by the arg parse.

default < config < commmand_line_args

### PR DESCRIPTION
**Pull request by Eugene-hu**
Add the functionality to load a config file using --neuron.config

### CODE REVIEW

File: miners/text/gpt2_exodus.py [hunk 1]
```diff
@@ -133,6 +135,29 @@ def config() -> 'bittensor.Config':
         bittensor.axon.add_args( parser )
         GPT2Nucleus.add_args( parser )
         SGMOERouter.add_args( parser )
+                
+        
+        
+        config_file_path = vars(parser.parse_known_args()[0])['neuron.config']
+        if config_file_path:
+            #loads config_file and updates defaults
+            if os.path.isabs(config_file_path) == False:
```
Comment (unconst) [inline miners/text/gpt2_exodus.py]: os.expanduser

File: miners/text/gpt2_exodus.py [hunk 2]
```diff
@@ -133,6 +135,29 @@ def config() -> 'bittensor.Config':
...
+                config_file_path = os.getcwd() + '/' + config_file_path
+
+                
+            try:
+                with open(config_file_path) as f:
+                    params_config = yaml.safe_load(f)
+                    params_config = munch.munchify(params_config)
```
Comment (unconst) [inline miners/text/gpt2_exodus.py]: can we use bittensor.Config

Comment (unconst) [inline miners/text/gpt2_exodus.py]: This is not needed

**Review by unconst — APPROVED** [review]

File: miners/text/gpt2_exodus.py [hunk 3]
```diff
@@ -133,6 +134,28 @@ def config() -> 'bittensor.Config':
...
+                
+            try:
+                with open(config_file_path) as f:
+                    params_config = yaml.safe_load(f)
+                    
+                    print('Config File Detected at' ,config_file_path, ', updating defaults')
+                    parser.set_defaults(**params_config)
+                    
+            except Exception as E:
+                print('Error in loading:', E, ', using default parser settings')
```
Comment (shibshib) [inline miners/text/gpt2_exodus.py]: Looks great @Eugene-hu , but can we switch the printing formatting? Throughout the code we use the `'{}'.format()` format. For example, `print("Error in loading {} using default parser settings".format(e))`. It's more python-3 :) 

Also, let's avoid using capital letters, best use `Exception as e`.

Comment (Eugene-hu) [reply miners/text/gpt2_exodus.py]: Fixed!

**Pull request merged**